src/mam4xx/water_uptake.hpp

```diff
@@ -429,7 +429,102 @@ void modal_aero_wateruptake_dryaer(
   }
 }
 
+KOKKOS_INLINE_FUNCTION
+void modal_aero_water_uptake_dr_b4_wetdnes(
+    int nspec_amode[AeroConfig::num_modes()],
+    Real specdens_amode[maxd_aspectype], Real spechygro[maxd_aspectype],
+    int lspectype_amode[maxd_aspectype][AeroConfig::num_modes()],
+    Real state_q[nvars], Real temperature, Real pmid, Real cldn,
+    Real dgncur_a[AeroConfig::num_modes()],
+    Real dgncur_awet[AeroConfig::num_modes()],
+    Real wetvol[AeroConfig::num_modes()], Real wtrvol[AeroConfig::num_modes()],
+    Real drymass[AeroConfig::num_modes()],
+    Real specdens_1[AeroConfig::num_modes()]) {
+
+  //----------------------------------------------------------------------------
+  // retreive aerosol properties
+
+  Real hygro[AeroConfig::num_modes()];
+  Real naer[AeroConfig::num_modes()];
+  Real dryrad[AeroConfig::num_modes()];
+  Real dryvol[AeroConfig::num_modes()];
+  Real rhcrystal[AeroConfig::num_modes()];
+  Real rhdeliques[AeroConfig::num_modes()];
+
+  modal_aero_water_uptake_dryaer(nspec_amode, specdens_amode, spechygro,
+                                 lspectype_amode, state_q, dgncur_a, hygro,
+                                 naer, dryrad, dryvol, drymass, rhcrystal,
+                                 rhdeliques, specdens_1);
+
+  // ----------------------------------------------------------------------------
+  // estimate clear air relative humidity using cloud fraction
+  Real rh;
+  modal_aero_water_uptake_rh_clearair(temperature, pmid, state_q[0], cldn, rh);
+
+  //----------------------------------------------------------------------------
+  // compute wet aerosol properties
+
+  // compute aerosol wet radius, volume, diameter and aerosol water
+  Real wetrad[AeroConfig::num_modes()];
+  Real qaerwat[AeroConfig::num_modes()];
+  modal_aero_water_uptake_wetaer(rhcrystal, rhdeliques, dgncur_a, dryrad, hygro,
+                                 rh, naer, dryvol, wetrad, wetvol, wtrvol,
+                                 dgncur_awet, qaerwat);
+}
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_water_uptake_dr(
```

Comment:
Does this function corresponds to the subroutine `modal_aero_wateruptake_dr` from [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/67ae6cd6f8b8381c3162b9071fe39754ab95fce7/components/eam/src/chemistry/utils/modal_aero_wateruptake.F90#L127).

Reply:
It does. I've added comments to make this clear.